### ISSUE TITLE
Module Generator plugin settings are over written when project is loaded

### ISSUE THREAD

**Issue by IrksomeZeus** [ThrowTheSwitch/Ceedling#291]
The Module generator settings defined in the _project.yml_ file are being overwritten by the values in _<project_root>\vendor\ceedling\plugins\module_generator\config\module_generator.yml_

I don't think this is the intended functionality but if it is please feel free to stop reading can close this issue. 

#### __Ceedling Version__
  Ceedling:: 0.28.3
CException:: 1.3.1.18
     CMock:: 2.4.6.217
     Unity:: 2.4.3.122

To easily demonstrate this, I put the following configuration in my _project.yml_ file:
```yaml
:module_generator:
  :project_root: ./
  :source_root: src/project_yml/
  :test_root: test/project_yml/
```

and this in the _<project_root>\vendor\ceedling\plugins\module_generator\config\module_generator.yml_:
```yaml
:module_generator:
  :project_root: ./
  :source_root: src/yml_mod_gen_config/
  :test_root: test/yml_mod_gen_config/
```

then running `ceedling module:create[mymod]` in the project directory (same dir as _project.yml_) the output shows that the project settings were not used.

```
\my_project>ceedling module:create[mymod]
\my_project>ruby vendor\ceedling\bin\ceedling module:create[mymod]
File src/yml_mod_gen_config/mymod.c created
File src/yml_mod_gen_config/mymod.h created
File test/yml_mod_gen_config/test_mymod.c created
Generate Complete
```

From doing some quick debugging,
it appears the issue is in `\vendor\ceedling\lib\ceedling\configurator.rb`
on line 186 where the config settings for each plugin are read in and merged with the project settings - before this point, the config_hash contains the _project.yml_ values for `[:module_generator][:value][:source_root]` and `[:module_generator][:value][:test_root]`

```ruby
    config_plugins.each do |plugin|
      config.deep_merge!( @yaml_wrapper.load(plugin) )
    end
```

**Comment by IrksomeZeus** [comment]
Just realized that there is a proposed fix for this here #215

**Comment by art-of-dom** [comment]
this should have been fixed by #365

**Comment by mvandervoord** [comment]
Thanks so much @art-of-dom -- as usual, your help has been invaluable!
